### PR TITLE
[4.4] Introduce local variables to replace repeating method calls

### PR DESCRIPTION
**Pull request by joomdonation**
Pull Request for Issue # .

### Summary of Changes
This PR introduces local variables `$app` and `$db` to replace the repeating calls `$this->getApplication()` and `$this->getDatabase()`. phpstorm has a built-in function called Refactor -> Introduce Variable allows us to do this quickly. We do not have a clear rule for this, but I think doing that would make code more cleaner and a little faster.

### Testing Instructions
1. Use Joomla 4.4
2. Apply patch
3. Try to login, check on Remember Me checkbox, make sure it is working OK
4. Code review would be great, too.
### Actual result BEFORE applying this Pull Request
Works

### Expected result AFTER applying this Pull Request
Works with cleaner code and a little faster.

### Link to documentations
Please select:
- [ ] Documentation link for docs.joomla.org: <link>
- [x] No documentation changes for docs.joomla.org needed

- [ ] Pull Request link for manual.joomla.org: <link>
- [x] No documentation changes for manual.joomla.org needed

### CODE REVIEW

File: plugins/authentication/cookie/src/Extension/Cookie.php [hunk 1]
```diff
@@ -104,29 +105,30 @@ public function onUserAuthenticate($credentials, $options, &$response)
         $now    = time();
 
         // Remove expired tokens
-        $query = $this->getDatabase()->getQuery(true)
-            ->delete($this->getDatabase()->quoteName('#__user_keys'))
-            ->where($this->getDatabase()->quoteName('time') . ' < :now')
+        $db = $this->getDatabase();
+        $query    = $db->getQuery(true)
```
Comment (sandewt) [inline plugins/authentication/cookie/src/Extension/Cookie.php]: spaces: 
$query = $db->getQuery(true)

File: plugins/authentication/cookie/src/Extension/Cookie.php [hunk 2]
```diff
@@ -224,24 +226,26 @@ public function onUserAuthenticate($credentials, $options, &$response)
     public function onUserAfterLogin($options)
     {
         // No remember me for admin
-        if ($this->getApplication()->isClient('administrator')) {
+        $app = $this->getApplication();
+        if ($app->isClient('administrator')) {
             return false;
         }
 
+        $db       = $this->getDatabase();
```
Comment (sandewt) [inline plugins/authentication/cookie/src/Extension/Cookie.php]: spaces:
$db = $this->getDatabase();

File: plugins/authentication/cookie/src/Extension/Cookie.php [hunk 3]
```diff
@@ -259,14 +263,14 @@ public function onUserAfterLogin($options)
 
             do {
                 $series = UserHelper::genRandomPassword(20);
-                $query  = $this->getDatabase()->getQuery(true)
-                    ->select($this->getDatabase()->quoteName('series'))
-                    ->from($this->getDatabase()->quoteName('#__user_keys'))
-                    ->where($this->getDatabase()->quoteName('series') . ' = :series')
+                $query  = $db->getQuery(true)
```
Comment (sandewt) [inline plugins/authentication/cookie/src/Extension/Cookie.php:268]: ~spaces:
$query = $db->getQuery(true)~

[EDIT]

File: plugins/authentication/cookie/src/Extension/Cookie.php [hunk 4]
```diff
@@ -376,19 +381,20 @@ public function onUserAfterLogout($options)
         $series = $filter->clean($cookieArray[1], 'ALNUM');
 
         // Remove the record from the database
-        $query = $this->getDatabase()->getQuery(true)
-            ->delete($this->getDatabase()->quoteName('#__user_keys'))
-            ->where($this->getDatabase()->quoteName('series') . ' = :series')
+        $db = $this->getDatabase();
+        $query    = $db->getQuery(true)
```
Comment (sandewt) [inline plugins/authentication/cookie/src/Extension/Cookie.php]: spaces:
$query = $db->getQuery(true)

**Comment by laoneo** [comment]
I'm not sure if there should be a blank line before the if blocks. Beside that it looks good! Thanks.

File: plugins/authentication/cookie/src/Extension/Cookie.php [hunk 5]
```diff
@@ -67,18 +67,19 @@ public function onPrivacyCollectAdminCapabilities()
     public function onUserAuthenticate($credentials, $options, &$response)
     {
         // No remember me for admin
-        if ($this->getApplication()->isClient('administrator')) {
+        $app = $this->getApplication();
```
Comment (laoneo) [inline plugins/authentication/cookie/src/Extension/Cookie.php:69]: Can you put that above the comment, as the comment belongs to the if block? Please on all occurrences.

Comment (joomdonation) [reply plugins/authentication/cookie/src/Extension/Cookie.php:69]: Thanks. It's done.

File: plugins/authentication/cookie/src/Extension/Cookie.php (hunk 3 above)
Comment (joomdonation) [reply plugins/authentication/cookie/src/Extension/Cookie.php:268]: Thanks @sandewt for reviewing.

**Comment by laoneo** [comment]
Thanks!

**Pull request merged**